File: 승준/20230704/rainwater_14719.java
import java.io.*;
import java.util.*;

public class rainwater_14719 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int h = Integer.parseInt(st.nextToken());
        int w = Integer.parseInt(st.nextToken());

        List<Integer> blocks = new ArrayList<>();
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < w; i++){
            blocks.add(Integer.parseInt(st.nextToken()));
        }

        int result = calculater_rainwater(h, w, blocks);
        System.out.println(result);
    }

    public static int calculater_rainwater (int h, int w, List<Integer> blocks) {
        int total = 0;
        for (int i = 1; i < w-1; i++){
            int left_max = Collections.max(blocks.subList(0, i));
            int right_max = Collections.max(blocks.subList(i+1, blocks.size()));

            if (blocks.get(i) < left_max && blocks.get(i) < right_max) {
                total += Math.min(left_max, right_max) - blocks.get(i);
            }
        }
        return total;
    }
}

File: 승준/20230704/rainwater_14719.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

function calculateRainwater(h, w, blocks) {
  let total = 0;
  for (let i = 1; i < w - 1; i++) {
    const leftMax = Math.max(...blocks.slice(0, i));
    const rightMax = Math.max(...blocks.slice(i + 1));

    if (blocks[i] < leftMax && blocks[i] < rightMax) {
      total += Math.min(leftMax, rightMax) - blocks[i];
    }
  }
  return total;
}

rl.on("line", (line) => {
  const input = line.split(" ");
  const h = parseInt(input[0]);
  const w = parseInt(input[1]);

  rl.on("line", (line) => {
    const blocks = line.split(" ").map(Number);
    const result = calculateRainwater(h, w, blocks);
    console.log(result);
    rl.close();
  });
});
